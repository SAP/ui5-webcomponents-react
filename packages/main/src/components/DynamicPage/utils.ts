export const DynamicPageCssVariables = {
  headerDisplay: '--_ui5wcr_DynamicPage_header_display',
  titleFontSize: '--_ui5wcr_DynamicPage_title_fontsize'
};
