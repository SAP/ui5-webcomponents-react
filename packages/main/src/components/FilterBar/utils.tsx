import { cloneElement } from 'react';

const inputTagNames = new Set([
  'UI5-COMBOBOX',
  'UI5-DATE-PICKER',
  'UI5-DATERANGE-PICKER',
  'UI5-DATETIME-PICKER',
  'UI5-DURATION-PICKER',
  'UI5-INPUT',
  'UI5-MULTI-INPUT',
  'UI5-RATING-INDICATOR',
  'UI5-STEP-INPUT',
  'UI5-TIME-PICKER'
]);

export const filterValue = (ref, child) => {
  const tagName = ref.tagName;
  let filterItemProps = {};
  if (inputTagNames.has(tagName)) {
    filterItemProps = { value: ref.value };
  }
  if (tagName === 'UI5-SELECT' || tagName === 'UI5-MULTI-COMBOBOX') {
    const selectedIndices = Array.from(ref.children as HTMLCollectionOf<any>)
      .map((item, index) => (item.selected ? index : false))
      .filter((el) => el !== false);
    const selectedIndicesSet = new Set(selectedIndices);
    const options = child.props.children.props.children?.map((item, index) => {
      if (selectedIndicesSet.has(index)) {
        return cloneElement(item, { selected: true });
      }
      return cloneElement(item, { selected: false });
    });
    filterItemProps = { children: options };
  }
  if (tagName === 'UI5-SWITCH' || tagName === 'UI5-CHECKBOX') {
    filterItemProps = { checked: ref.checked };
  }
  return filterItemProps;
};

export const renderSearchWithValue = (renderSearchElement, searchValue) => {
  return cloneElement(renderSearchElement, { value: searchValue ?? renderSearchElement?.props?.value });
};
