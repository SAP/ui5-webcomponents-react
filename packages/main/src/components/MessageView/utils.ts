import iconAlert from '@ui5/webcomponents-icons/dist/alert.js';
import iconError from '@ui5/webcomponents-icons/dist/error.js';
import iconInformation from '@ui5/webcomponents-icons/dist/information.js';
import iconSysEnter from '@ui5/webcomponents-icons/dist/sys-enter-2.js';
import { ValueState } from '../../enums/ValueState';

export const getIconNameForType = (type: ValueState | keyof typeof ValueState): string => {
  switch (type) {
    case ValueState.Error:
    case 'Error':
      return iconError;
    case ValueState.Success:
    case 'Success':
      return iconSysEnter;
    case ValueState.Warning:
    case 'Warning':
      return iconAlert;
    default:
      return iconInformation;
  }
};
