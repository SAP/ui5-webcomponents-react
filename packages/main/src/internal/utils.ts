import { Children, Fragment, ReactNode } from 'react';

export function flattenFragments(children: ReactNode | ReactNode[], depth = 1): ReactNode[] {
  const flatChildren = [];
  if (Array.isArray(children)) {
    return children;
  }
  const removeFragments = (element, level = 0) => {
    if (!element) {
      return;
    }
    if (element.type === Fragment && level <= depth) {
      Children.toArray(element.props?.children).forEach((item) => {
        removeFragments(item, level + 1);
      });
    } else {
      flatChildren.push(element);
    }
  };
  removeFragments(children);

  return flatChildren;
}
