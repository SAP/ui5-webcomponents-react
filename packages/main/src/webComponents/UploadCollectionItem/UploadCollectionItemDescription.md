A component to be used within the `UploadCollection`.
