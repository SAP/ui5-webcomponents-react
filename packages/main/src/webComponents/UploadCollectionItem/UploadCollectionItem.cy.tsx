import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { UploadCollectionItem } from './index';

describe('UploadCollectionItem', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<UploadCollectionItem />);
    cy.get('[ui5-upload-collection-item]').should('exist');
  });

  cypressPassThroughTestsFactory(UploadCollectionItem);
});
