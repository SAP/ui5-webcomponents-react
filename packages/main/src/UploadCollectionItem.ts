import { deprecationNotice } from '@ui5/webcomponents-react-base';
import type { UploadCollectionItemDomRef, UploadCollectionItemPropTypes } from './webComponents/UploadCollectionItem';
import { UploadCollectionItem } from './webComponents/UploadCollectionItem';

deprecationNotice(
  'UploadCollectionItem',
  `importing from '@ui5/webcomponents-react/dist/UploadCollectionItem' is deprecated. Please use "import { UploadCollectionItem } from '@ui5/webcomponents-react';" instead.`
);

export { UploadCollectionItem };

export type { UploadCollectionItemPropTypes, UploadCollectionItemDomRef };
