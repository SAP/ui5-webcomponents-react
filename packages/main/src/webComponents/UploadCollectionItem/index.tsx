import '@ui5/webcomponents-fiori/dist/UploadCollectionItem.js';
import { ReactNode } from 'react';
import { UploadState, ListItemType } from '../../enums';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5CustomEvent } from '../../interfaces/Ui5CustomEvent';
import { Ui5DomRef } from '../../interfaces/Ui5DomRef';
import { withWebComponent } from '../../internal/withWebComponent';

interface UploadCollectionItemAttributes {
  /**
   * Disables the delete button.
   */
  disableDeleteButton?: boolean;
  /**
   * Holds an instance of `File` associated with this item.
   */
  file?: File;
  /**
   * The name of the file.
   */
  fileName?: string;
  /**
   * If set to `true` the file name will be clickable and it will fire `onFileNameClick` event upon click.
   */
  fileNameClickable?: boolean;
  /**
   * Hides the retry button when `uploadState` property is `Error`.
   */
  hideRetryButton?: boolean;
  /**
   * Hides the terminate button when `uploadState` property is `Uploading`.
   */
  hideTerminateButton?: boolean;
  /**
   * The upload progress in percentage.
   *
   * **Note:** Expected values are in the interval \[0, 100\].
   */
  progress?: number;
  /**
   * If set to `Uploading` or `Error`, a progress indicator showing the `progress` is displayed. Also if set to `Error`, a refresh button is shown. When this icon is pressed `onRetry` event is fired. If set to `Uploading`, a terminate button is shown. When this icon is pressed `onTerminate` event is fired.
   */
  uploadState?: UploadState | keyof typeof UploadState;
  /**
   * Defines the visual indication and behavior of the list items. Available options are `Active` (by default), `Inactive` and `Detail`.
   *
   * **Note:** When set to `Active`, the item will provide visual response upon press and hover, while with type `Inactive` and `Detail` - will not.
   */
  type?: ListItemType | keyof typeof ListItemType;
  /**
   * Defines the selected state of the `ListItem`.
   */
  selected?: boolean;
}

export interface UploadCollectionItemDomRef extends UploadCollectionItemAttributes, Ui5DomRef {}

export interface UploadCollectionItemPropTypes extends UploadCollectionItemAttributes, CommonProps {
  /**
   * Hold the description of the `UploadCollectionItem`. Will be shown below the file name.
   */
  children?: ReactNode | ReactNode[];
  /**
   * A thumbnail, which will be shown in the beginning of the `UploadCollectionItem`.
   *
   * **Note:** Use `Icon` or `img` for the intended design.
   *
   * __Note:__ When passing a custom React component to this prop, you have to make sure your component reads the `slot` prop and appends it to the most outer element of your component.
   * Learn more about it [here](https://sap.github.io/ui5-webcomponents-react/?path=/docs/knowledge-base-handling-slots--page).
   */
  thumbnail?: ReactNode;
  /**
   * Fired when the file name is clicked.
   *
   * **Note:** This event is only available when `fileNameClickable` property is `true`.
   */
  onFileNameClick?: (event: Ui5CustomEvent<UploadCollectionItemDomRef>) => void;
  /**
   * Fired when the `fileName` property gets changed.
   *
   * **Note:** An edit button is displayed on each item, when the `UploadCollectionItem` `type` property is set to `Detail`.
   */
  onRename?: (event: Ui5CustomEvent<UploadCollectionItemDomRef>) => void;
  /**
   * Fired when the retry button is pressed.
   *
   * **Note:** Retry button is displayed when `uploadState` property is set to `Error`.
   */
  onRetry?: (event: Ui5CustomEvent<UploadCollectionItemDomRef>) => void;
  /**
   * Fired when the terminate button is pressed.
   *
   * **Note:** Terminate button is displayed when `uploadState` property is set to `Uploading`.
   */
  onTerminate?: (event: Ui5CustomEvent<UploadCollectionItemDomRef>) => void;
  /**
   * Fired when the user clicks on the detail button when type is `Detail`.
   */
  onDetailClick?: (event: Ui5CustomEvent<UploadCollectionItemDomRef>) => void;
}

/**
 * A component to be used within the `UploadCollection`.
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/UploadCollectionItem" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const UploadCollectionItem = withWebComponent<UploadCollectionItemPropTypes, UploadCollectionItemDomRef>(
  'ui5-upload-collection-item',
  ['file', 'fileName', 'progress', 'uploadState', 'type'],
  ['disableDeleteButton', 'fileNameClickable', 'hideRetryButton', 'hideTerminateButton', 'selected'],
  ['thumbnail'],
  ['file-name-click', 'rename', 'retry', 'terminate', 'detail-click']
);

UploadCollectionItem.displayName = 'UploadCollectionItem';

UploadCollectionItem.defaultProps = {
  file: null,
  progress: 0,
  uploadState: UploadState.Ready,
  type: ListItemType.Active
};

export { UploadCollectionItem };
