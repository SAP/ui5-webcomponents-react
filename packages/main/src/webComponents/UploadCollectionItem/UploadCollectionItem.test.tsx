import { render } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { UploadCollectionItem } from '@ui5/webcomponents-react/dist/UploadCollectionItem';
import React from 'react';

describe('UploadCollectionItem', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(<UploadCollectionItem />);
    expect(asFragment()).toMatchSnapshot();
  });
  createCustomPropsTest(UploadCollectionItem);
});
