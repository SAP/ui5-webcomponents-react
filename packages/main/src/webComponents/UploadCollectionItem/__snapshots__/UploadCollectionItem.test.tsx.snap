// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadCollectionItem Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-upload-collection-item
    progress="0"
    type="Active"
    ui5-upload-collection-item=""
    upload-state="Ready"
  />
</DocumentFragment>
`;
